### ISSUE TITLE
Ubuntu 19.04 - keybindings don't work

### ISSUE THREAD

**Issue by jendakol** [Keruspe/GPaste#272]
Hi,
I've downloaded gpaste 3.32.0 tarball (because it's still not available in official repos) and run:
```
sudo apt build-dep gpaste
./configure --sysconfdir=/etc
make
sudo make install
sudo glib-compile-schemas /usr/share/glib-2.0/schemas/
```
The gpaste itself works, shell extension displays it's icon in the top bar, clipboard catching works. 
 
BUT: keyboard shortcuts don't. When I try to define a shortcut for "Display the history" and press `<Alt>` the whole settings window disappear. Already defined shortcuts (defaults) don't work too - e.g. displaying the UI.  
When I run the daemon in debug mode:
```
jenda@placka: /tmp/gpaste-3.32.0 » env G_MESSAGES_DEBUG=all /usr/libexec/gpaste/gpaste-daemon                                                                                       [22:31:13]
(gpaste-daemon:11609): GLib-GIO-DEBUG: 22:31:14.373: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
(gpaste-daemon:11609): Gtk-DEBUG: 22:31:14.401: Connecting to session manager
(gpaste-daemon:11609): GLib-GIO-DEBUG: 22:31:14.423: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
(gpaste-daemon:11609): dconf-DEBUG: 22:31:14.423: watch_fast: "/org/gnome/GPaste/" (establishing: 0, active: 0)
(gpaste-daemon:11609): dconf-DEBUG: 22:31:14.423: watch_fast: "/org/gnome/shell/" (establishing: 0, active: 0)
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.423: history: elect biggest
(gpaste-daemon:11609): dconf-DEBUG: 22:31:14.423: watch_established: "/org/gnome/GPaste/" (establishing: 1)
(gpaste-daemon:11609): dconf-DEBUG: 22:31:14.423: watch_established: "/org/gnome/shell/" (establishing: 1)
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.428: CLIPBOARD: set text
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.429: atoms init
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: item: set state: 1
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: history: selected
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: clipboards-manager: select
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: PRIMARY: select item
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: PRIMARY: set text
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: CLIPBOARD: select item
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: CLIPBOARD: set text
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.510: history: elect biggest
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: PRIMARY: owner change
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: notify
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: targets ready
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: text ready
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: notify finish
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: CLIPBOARD: owner change
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: notify
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: targets ready
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: text ready
(gpaste-daemon:11609): GPaste-DEBUG: 22:31:14.511: clipboards-manager: notify finish

(gpaste-daemon:11609): GPaste-WARNING **: 22:31:14.513: Couldn't grab keybindings with gnome-shell: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: Type of message, “(a(su))”, does not match expected type “(a(suu))”

(gpaste-daemon:11609): GPaste-WARNING **: 22:31:14.513: Couldn't grab keybindings with gnome-shell: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: Type of message, “(a(su))”, does not match expected type “(a(suu))”
```
I have:
```
GNOME Shell 3.32.0
5.0.0-13-generic
```
Did I miss something/do something wrong?

**Comment by pseudo-morph** [comment]
I have the same issue with Ubuntu 19.04 and GPaste 3.30.2 from the repo.

**Comment by Keruspe** [comment]
GPaste 3.30 is not compatible with gnome-shell 3.32

**Comment by pseudo-morph** [comment]
Ah, that might be the problem then. :)

**Comment by Keruspe** [comment]
There’s an issue open at Ubuntu to update their packages

**Comment by thiagopa** [comment]
Installed gpaste-3.32.0 from source and keybinds don't work on Ubuntu 19.04 with Gnome Shell 3.32.1

dbus-monitor interface=org.gnome.Shell doesn't show any shortcut assigned to Gpaste (but other shortcuts that don't belong to Gpaste work and appear on it)

**Comment by Keruspe** [comment]
What does « gpaste-client daemon-version » tell you?

**Comment by jendakol** [comment]
Some hanging libraries in my system (`libgpaste11` and some others) caused the deamon to not update and it was stuck on version `3.30.2`. I cleared them, installed everything again from sources and now it's `3.32.0` and shortcuts work :tada:  
However, the story doesn't end completely. Not that it would block me currently but I'm still unable to redefine shortcuts - the problem with settings window disappearing after `<Alt>` press remains.

**Comment by thiagopa** [comment]
Nice tip Keruspe, managed to fix it. 

Initially it was :

gpaste-client daemon-version
3.30.2

But it should be 3.32.0.

So what did I do to fix it was to remove the ubuntu package with:

sudo apt-get remove gpaste

Then installed from my previous build with:

sudo make install 
sudo glib-compile-schemas /usr/share/glib-2.0/schemas/

But the story wasn't over, had to refresh the daemon from memory with:

sudo gpaste-client daemon-reexec

and now it is:

gpaste-client daemon-version
3.32.0

Shortcut works perfectly, Thank you!

**Comment by thiagopa** [comment]
@jendakol I can redefine the keys from the official gnome shell keyboard shortcuts settings, there's a dedicated space for GPaste.

**Comment by jendakol** [comment]
@thiagopa I know, did found it too :-) As I said, it's currently not a problem for me, since I have all set. However, the behavior is not what it should be... Does it happen to you too or it's just me? (read original post for more info)

**Comment by thiagopa** [comment]
Yes, tried to change the keybindings again and now it doesn't work anymore. =(

gpaste-daemon shows this message though:

(gpaste-daemon:18270): GPaste-WARNING **: 15:51:59.966: Couldn't grab keybinding: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: Type of message, “(su)”, does not match expected type “(suu)”

**Comment by thiagopa** [comment]
Killed and restarted the daemon and now it works again, better not mess with the keybindings again as there's some bug that prevents the shortcut to be recognized if I try to set the bindings from the gpaste own window. (From Gnome Configuration works like a charm though)

**Comment by jendakol** [comment]
I know this warning very well ;-) Glad to hear it's not just my problem. Now I'm curious what the cause could be...

**Comment by Keruspe** [comment]
FWIW, I just released 3.34.3 and 3.36.2 with fixes for this. Is a 3.32 release also needed? (and would it hit ubuntu maintainers or just be ignored?)

**Comment by Keruspe** [comment]
3.32.1 is on its way
